proposals/0311-task-locals.md

```diff
@@ -1194,6 +1153,36 @@ As discussed in the tracing use-case section, the ability to express `@Logged` o
 
 Such annotations depend on the arrival of [Function Wrappers](https://forums.swift.org/t/prepitch-function-wrappers/33618) or a similar feature to them, which currently are not being actively worked on, however we definitely have in the back of our minds while designing this proposal.
 
+### Language features to avoid nesting with `withValue`
+
+It is necessary for task-local correctness to only bind values for a given scope. 
+
+This scoping rule is enforced by the only API to bind a task-local value being the `withValue() { ... }` function. The function essentially does two operations: 
+
+- pushes a new binding onto the task-local bindings stack in the current task
+- (executes the user provided `body` closure)
+- and pops the binding from the task-local bindings stack of the current task
+
+This structure allows us to use task-local allocation safely, and also ensure that no lingering values ever "leak" the scope where they were defined.
+
+It is, however, slightly cumbersome on a source level to have to indent code only in order to get this property when in reality most of the time a binding is going to be set for the entirety (or remaining part) of the current function.
+
+It Swift were to gain some "`using`" mechanism, that would encapsulate the pattern of doing one "start" operation now, and an "end" operation at scope exit. That would bee a very general feature with broad applicability. The same mechanism could be used by task-locals rather than resorting to nesting. 
```

Comment:
```suggestion
If Swift were to gain some "`using`" mechanism, that would encapsulate the pattern of doing one "start" operation now and an "end" operation at scope exit. That would be a very general feature with broad applicability. The same mechanism could be used by task-locals rather than resorting to nesting. 
```